src/base/bittorrent/torrentimpl.cpp

```diff
@@ -142,11 +148,21 @@ namespace
                     {
                         trackerEndpoint.status = TrackerEntry::NotContacted;
                     }
-                    trackerEntry.endpoints.append(trackerEndpoint);
 
-                    trackerEntry.numSeeds = std::max(trackerEntry.numSeeds, infoHash.scrape_complete);
-                    trackerEntry.numLeeches = std::max(trackerEntry.numLeeches, infoHash.scrape_incomplete);
-                    trackerEntry.numDownloaded = std::max(trackerEntry.numDownloaded, infoHash.scrape_downloaded);
+                    const QString trackerMessage = QString::fromStdString(infoHash.message);
+                    const QString errorMessage = QString::fromLocal8Bit(infoHash.last_error.message().c_str());
+                    trackerEndpoint.message = (!trackerMessage.isEmpty() ? trackerMessage : errorMessage);
```

Comment:
what about using: 
```c++
trackerEndpoint.message = (!errorMessage.isEmpty() ? errorMessage : trackerMessage);
```

Reply:
We should prefer message from tracker, shouldn't we?
In case there is network error "tracker message" is empty and we use lower level error message. Otherwise we use the message returned from tracker.

Reply:
error messages will never be empty since we receive messages even from local endpoints like 127.0.0.1 or link local IPv6 addresses. So we must prefer the message sent from tracker rather than error message.